keyvaluestore/node.go

```diff
@@ -165,6 +168,7 @@ func (n *node) shiftRightmostElementToNext() error {
 
 	n.n--
 	next.n++
+	next.isDirty = true
```

Comment:
`n` should be marked as dirty too, right?

Reply:
This method only gets called after inserting an entry into n. It is already dirty at this point.